kbase-extension/static/kbase/js/common/cellComponents/tabs/jobStatus/jobStateListRow.js

```diff
@@ -1,157 +1,175 @@
 define([
     'bluebird',
-    'kb_common/html'
+    'kb_common/html',
+    'common/events',
+    'jquery'
 ], function(
     Promise,
-    html
+    html,
+    Events,
+    $
 ) {
     'use strict';
 
+    var container;
+
     const t = html.tag,
         div = t('div'),
         td = t('td'),
         span = t('span'),
-        cssBaseClass = 'kb-job-state';
+        a = t('a'),
+        i = t('i'),
+        cssBaseClass = 'kb-job-status',
+        events = Events.make();
 
     function createStateCell(jobState) {
-        let label, icon;
+        let label;
         switch (jobState) {
             case 'completed':
-                label = 'success';
-                icon = 'fa fa-check';
+                label = 'Success';
                 break;
             case 'created':
+            case 'estimating':
             case 'queued':
-                label = jobState;
-                icon = 'fa fa-angle-double-right';
+                label = 'Queued';
                 break;
             case 'running':
-                label = jobState;
-                icon = 'fa fa-spinner';
+                label = 'Running';
                 break;
             case 'error':
+                label = 'Failed';
+                break;
             case 'terminated':
-                label = jobState;
-                icon = 'fa fa-times';
+                label = 'Cancelled';
                 break;
             case 'does_not_exist':
-                label = 'does not exist';
-                icon = 'fa fa-question';
+                label = 'Does not exist';
                 break;
-            case 'estimating':
-            default:
-                label = jobState;
-                icon = 'fa fa-question';
         }
 
         return td({
-            class: `${cssBaseClass}__cell--status`,
+            class: `${cssBaseClass}__cell`
         }, [
             span({
-                class: `${icon} ${cssBaseClass}__icon--${jobState}`
+                class: `fa fa-circle ${cssBaseClass}__icon--${jobState}`
             }),
             ' ' + label
         ]);
     }
 
-    function createActionCell(jobState) {
+    function selectRow(e) {
+        let $currentRow = $(e.target).closest('tr');
+
+        let $allRows = $('.kb-job-status__row');
+
+        // unselect previously selected row
+        $allRows.removeClass(`${cssBaseClass}__row_selected`);
+        $allRows.find('.selected_log').css('display', 'none');
```

Comment:
is there a way to use this same pattern for both select and unselect so there's less duplication?

Reply:
Changed :)